core/src/main/java/io/questdb/cairo/pool/SqlCompilerPool.java

```diff
@@ -25,22 +22,27 @@ public final class SqlCompilerPool extends AbstractMultiTenantPool<SqlCompilerPo
     private final Rnd rnd = new Rnd();
 
     public SqlCompilerPool(CairoEngine engine) {
-        super(engine.getConfiguration(), MAX_POOL_SEGMENTS, INACTIVE_COMPILER_TTL_MILLIS);
+        super(engine.getConfiguration(), engine.getConfiguration().getSqlCompilerPoolMaxSegments(), 0L);
```

Comment:
TTL is now set to 0. Do we want to keep up to 32 * 10 SQL compilers around?

Reply:
Please, note, that since `releaseInactive()` is never called on `SqlCompilerPool`, this value could be set to pretty much anything. It is not used basically.

SqlCompiler is not the usual beast.
While table readers/writers could hang around for some time, SQLCompiler does not. And it really should not.
Usage pattern for SQLCompiler is:
```
try (SqlCompiler compiler = engine.getSqlCompiler()) {
    compiler.compile(sqlStatement, sqlExecutionContext);
}
```
SQLCompiler should be closed immediately after usage.
If not closed, it is a leak.

Reply:
> Please, note, that since `releaseInactive()` is never called on `SqlCompilerPool`, this value could be set to pretty much anything. It is not used basically.

My bad, I didn't check if `releaseInactive()` is actually called. Worth adding a clarifying comment on 0 TTL to avoid confusion in future.

Reply:
> SQLCompiler should be closed immediately after usage.
> If not closed, it is a leak.

The thing is that when a pool tenant is closed, the `SqlCompiler` is not closed and these objects are heavy enough.

Reply:
"Worth adding a clarifying comment on 0 TTL to avoid confusion" -> will do in next commit

Reply:
After talking to Vlad, instead of setting max pool size from config I will attempt to calculate the appropriate size based on number of worker threads in the actual QuestDB instance.
The idea is that no thread will/should use more than 1 SqlCompiler at a time, so if i get the number of worker threads potentially using SqlCompilers, and let's say, double it, that should be more than enough.
Obviously, this will be a much lower number than the current setting in the config.

Reply:
I recall some situations where we have nested compilation, so a single thread may be using multiple compilers. Need to refresh my memory on the exact scenario though.

Reply:
we don't use multiple compilers from one thread, nested compilation is supported by single compiler instance

Reply:
added the calculation in fdbb552f54bb145ff6f85feb3bab6530f6f1254b